src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -967,25 +966,13 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				this._messageTitleDisposable = this._processManager.onProcessTitle(title => this.setTitle(title ? title : '', TitleEventSource.Process));
 			}
 		}
-
-		if (platform.isWindows) {
-			this._processManager.ptyProcessReady.then(() => {
-				if (this._processManager.remoteAuthority) {
-					return;
-				}
-				this._xtermReadyPromise.then(xterm => {
-					if (!this._isDisposed && this._processManager && this._processManager.shellProcessId) {
-						this._windowsShellHelper = this._terminalInstanceService.createWindowsShellHelper(this._processManager.shellProcessId, xterm);
-						this._windowsShellHelper.onShellNameChange(title => {
-							this.setShellType(this.getShellType(title));
-							if (this.isTitleSetByProcess && !this._configHelper.config.experimentalUseTitleEvent) {
-								this.setTitle(title, TitleEventSource.Process);
-							}
-						});
-					}
-				});
-			});
-		}
+		this._processManager.onShellTypeChanged(type => {
+			// only set if it's changed
+			if (this._shellType !== type) {
```

Comment:
We don't actually need this since `setShellType` doesn't seem to fire any events, it just sets. If it did, we would want to move this conditional into `setShellType`.

Reply:
trying to avoid setting the title again if it hasn't changed

Reply:
It might actually be faster to just set it here due to branch prediction (extremely minor point though), but it would also be less code to avoid the if. We would only need to check this if `_shellType` was a getter that did something or if `setShellType` fired some on change event.